common/build/eslint-config-fluid/src/custom-rules/no-restricted-tags-imports.js

```diff
@@ -0,0 +1,161 @@
+		context.report({
+			loc: { line: 1, column: 0 },
+			message: `Invalid tag format in rule configuration: '{${tag}}'. Tags should start with '@'.`,
+			data: { tag },
+		});
+	} else {
+		return true;
+	}
+}
+
+/**
+ * Processes the tags array to ensure all tags are valid.
+ * @param tags - Array of tags to be processed.
+ * @param context - The context object provided by ESLint.
+ * @returns A set of validated tags.
+ */
+function processTags(tags, context) {
+	return new Set(tags.filter((tag) => isTagValid(tag, context)));
+}
+
+/**
+ * Validate the exceptions object, ensuring all tags and paths are valid.
+ * @param exceptions - The exceptions object from the rule configuration.
+ * @param context - The context object provided by ESLint.
+ * @returns - An object with tags as keys and Sets of paths as values.
+ */
+function processExceptions(exceptions, context) {
+	const processedExceptions = {};
+	Object.keys(exceptions).forEach((tag) => {
+		if (isTagValid(tag, context)) {
+			processedExceptions[tag] = new Set(exceptions[tag]);
+		}
+	});
+	return processedExceptions;
+}
+
+/**
+ * Resolves the path of an imported module to an absolute path.
+ *
+ * @param importPath - The path used in the import statement.
+ * @param currentFilePath - The absolute path of the current file.
+ * @return - The absolute path of the imported module.
+ */
+function resolveImportPath(importPath, currentFilePath) {
+	return path.isAbsolute(importPath)
+		? importPath
+		: path.resolve(path.dirname(currentFilePath), importPath);
+}
+
+module.exports = {
+	meta: {
+		type: "problem",
+		docs: {
+			description:
+				"This rule restricts imports from specified tags or non-public APIs. This to prevent accidental dependencies on internal, unstable or undocumented parts of the codebase.",
+			category: "Best Practices",
+		},
+		fixable: "code",
+		schema: [
+			{
+				type: "object",
+				properties: {
+					tags: {
+						type: "array",
+						items: { type: "string" },
+						uniqueItems: true,
+					},
+					exceptions: {
+						type: "object",
+						additionalProperties: {
+							type: "array",
+							items: { type: "string" },
+							uniqueItems: true,
+						},
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+		messages: {
+			importWithRestrictedTag: "Import with restricted tag found.",
+		},
+	},
+	create(context) {
+		const options = context.options[0] || {};
+		const restrictedTags = processTags(options.tags, context);
+		const exceptions = processExceptions(options.exceptions, context);
+		const project = new Project();
+
+		let tsConfigPath;
+		if (context.parserOptions.project) {
+			// Resolve the relative path to an absolute path
+			tsConfigPath = path.resolve(context.getCwd(), context.parserOptions.project);
+		} else {
+			// Fallback to a default/test path or handle absence of tsconfig.json as needed
+			tsConfigPath = path.join(__dirname, "../test/mockFiles/**/*.ts");
+		}
+		project.addSourceFilesAtPaths(tsConfigPath);
```

Comment:
Nit: it looks like we could move the instantiation of `project` down here and provide the tsconfigPath at construction.
https://ts-morph.com/setup/adding-source-files